Add plugin: Sync Emails

taskrobin opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/taskrobin/Obsidian-Plugin/

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

taskrobin commented:
Resolved required changed and pushed to repo under release 1.0.0. Please re-evaluate.

joethei commented:
["dice",](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L45)
Use a different icon than the default from the sample plugin
You can find all available icons and how to add your own here: https://docs.obsidian.md/Plugins/User+interface/Icons

[name: "Open TaskRobin Window",](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L56), [contentEl.createEl("h2", { text: "Setup Email Sync" });](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L168), [text: `Sync Emails`,](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L454), [text: "Sync Now",](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L507), [text: "Open Settings",](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L513), [text: "Delete Integration",](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L518), [.setName("Access Token")](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L755), [.setName("Root Directory")](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L770), ["Create Directory"](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L788), [.setName("Download Attachments")](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L813)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[const folderExists =](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L592-L595), [const folderExists =](https://github.com/taskrobin/Obsidian-Plugin/blob/d7693a329b772e4299981de76783255cb035d8ef/main.ts#L791-L792)
[Prefer the Vault API over the Adapter API](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Prefer+the+Vault+API+over+the+Adapter+API)

taskrobin commented:
Updated source code according to all comments. Ready for review @joethei

taskrobin commented:
Changes have been made. Ready for review @ObsidianReviewBot

joethei commented:
Just missed one.
The last line I flagged has not been changed.

taskrobin commented:
@joethei Do you mean the button casing issue? Just fixed the remaining ones and pushed. 
I think I have fixed all the adapter APIs, please let me know if something is still pending.

taskrobin commented:
Changes have been made. Ready for review @ObsidianReviewBot @joethei

taskrobin commented:
@ObsidianReviewBot please change PR status, this is ready for human review
